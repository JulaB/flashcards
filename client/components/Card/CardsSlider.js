import React from 'react';
import PropTypes from 'prop-types';
import Slider from 'components/Shared/Slider/Slider';
import Card from 'components/Card/Card';

const CardsSlider = ({ cards }) => (
  <Slider>
    {cards.map(card => <Card {...card.attributes} key={card.id} />)}
  </Slider>
);

CardsSlider.propTypes = {
  cards: PropTypes.arrayOf(PropTypes.object),
};

CardsSlider.defaultProps = {
  cards: [],
};
export default CardsSlider;
