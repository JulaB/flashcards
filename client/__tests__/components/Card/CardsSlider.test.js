import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import CardsSlider from 'components/Card/CardsSlider';

describe('<CardsSlider />', () => {
  it('renders the component with default props', () => {
    const wrapper = mount(<CardsSlider />);
    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.html()).toBe('');
  });

  it('renders slider with cards', () => {
    const cards = [
      { id: '1', attributes: { term: 'term1', definition: 'some definition' } },
      { id: '2', attributes: { term: 'term2', definition: 'some definition' } },
    ];
    const wrapper = shallow(<CardsSlider cards={cards} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
