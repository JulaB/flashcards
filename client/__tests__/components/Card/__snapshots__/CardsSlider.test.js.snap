// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardsSlider /> renders slider with cards 1`] = `
<Slider
  defaultActive={0}
>
  <Card
    definition="some definition"
    key="1"
    term="term1"
  />
  <Card
    definition="some definition"
    key="2"
    term="term2"
  />
</Slider>
`;
